### PR TITLE
Improve card features and functionality

### PR DESCRIPTION
**Pull request by thisisdano**
[Preview](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/dw-card-layout-grid/components/preview/card--test.html)→ 

This pull request does a number of things to improve the accessibility, styling, and functionality of cards.

## Overview
Cards are collected into card groups, and are coded as list items within unordered lists with the following HTML structure:

```
ul.usa-card-group
└─li.usa-card
│ └─ usa-card__container
│    ├─ usa-card__header
│    ├─ usa-card__media
│    ├─ usa-card__body
│    └─ usa-card__footer
└─li.usa-card
│ └─ usa-card__container
│    ├─ usa-card__header
│    ├─ usa-card__media
│    ├─ usa-card__body
│    └─ usa-card__footer
...
```

The `header`, `media`, `body`, and `footer` elements are all optional, but most cards would be expected to have a `header` and a `body`.

Card groups are functionally similar to **grid rows**, and each card is functionally similar to a **grid column**. This means that you can add grid (and flex) utilities to these classes to affect their output, like `usa-card-group.flex-align-start.grid-gap-1` and `usa-card.tablet:grid-col-6`.

The card component default is vertical orientation. Note that though the `media` element displays before the `header` element by default, it _always_ follows the `header` element in the markup.

#### Variant options

Class | Usage
--- | ---
`usa-card--flag` | Display in a horizontal ("flag") orientation at a specified width (`$theme-card-flag-min-width`)
`usa-card--header-first` | Displays the header element before the media element.
`usa-card--media-right` | In combination with `usa-card--flag`, sets the media element on the right. (Flag cards display media on the left by default.)
`usa-card__media--indent` | Indents the media element so it doesn't extend to the edge of the card.
`usa-card__media--set-aspect` | Sets a fixed aspect ratio on the card media. The default is 16x9, but this can be changed by adding an `add-aspect` utility to the media element, like `usa-card__media--set-aspect.add-aspect-1x1`
`usa-card__media--exdent` | Extends the media element out over the card border. Useful for light-bordered cards.
`usa-card__body--exdent` | Extends the body element out over the card border. Useful for light-bordered cards.
`usa-card__footer--exdent` | Extends the footer element out over the card border. Useful for light-bordered cards.

#### Settings options
Setting | Usage | Default
--- | --- | ---
`$theme-card-border-color` | Border color | `"base-lighter"`
`$theme-card-border-radius` | Border radius | `"lg"`
`$theme-card-border-width` | Border width | `2px`
`$theme-card-gap` | Gap between cards in a card group | 2
`$theme-card-flag-min-width` | Width at which flag cards change to horizontal layout | `"tablet"`
`$theme-card-flag-image-width` | Fixed image width in the flag variant | `"card-lg"`
`$theme-card-font-family` | Font family for the card body | `"body"`
`$theme-card-header-typeset` | Family, size, and line height of the heading | `"heading", "lg", 2`
`$theme-card-margin-bottom` | Margin below the card | `4`
`$theme-card-padding-perimeter` | Padding between card elements and the border of the card | `3`
`$theme-card-padding-y` | Vertical padding between card elements | `2`

## Changes in this pull request
**Cards are list items within a card group unordered list.** This not only groups cards logically, but provides useful functionality to screen readers — that the card group contains `x` number of elements, and readers can jump between them. The card group class is `usa-card-group`. ([Reference: Inclusive components](https://inclusive-components.design/cards/))

**Cards always begin with the header element.** This assures that any card content, including media is accessible to readers navigating by header through the document. 

**Card size and layout is controlled with grid utilities.** The `usa-card-group` block is functionally a wrap-able `grid-row` and each individual `usa-card` is a `grid-col`. This means that you can use a grid utility like `tablet:grid-col-4` to set a four-column size on a card at `tablet` width. You can also set custom grid gaps on a `usa-card-group` using the `grid-gap` utilities. Since the USWDS grid is based on flexbox, you can use flex positioning utilities on the card group. The default alignment is stretch (this aligns the top and bottom of each card in a row), but the `flex-align-start` utility can set the alignment to the top of the row. 

**Uses the same markup for all card variants.** This no longer requires special markup for the `--flag` variant or a `--media` variant for cards that include media. Now media works properly without a variant, and a card can be changed to flag orientation with a class only.

**Decouples the vertical padding from the perimeter padding.** Now, designers can set independent values for the padding around the perimeter of the card (`$theme-card-padding-perimeter`) and the vertical padding between elements (`$theme-card-padding-y`)

**Removes the `--raised` variant in favor of using utility classes.** Since the `--raised` variant only added a `shadow` to the card container, this could achieved using a `shadow-n` utility class instead.

**Images set to `object-fit: cover`.** This allows the image to more reliably adapt to its card container, no matter the width or context.

**Flag media has a fixed width, with a default of stretching the height of the card.** This provides a more consistent appearance for flag cards.

#### Related fixes
- Added `auto` as an acceptable value for `margin-y`
- Removed `max-width: 100%` from `add-aspect` utility/mixin
- Allowed `typeset()` mixin to accept variable lists, like `$foo: a, b, c → typeset($foo)`

## New variants
`usa-card--header-first`: Reorders the card so the header appears before the media. This is tricky to accomplish only with utility classes because it's complicated by the border radius and possible exdent.

Screen Shot 2020-04-27 at 3 28 29 PM

> Above: A standard card on the left, and a `--header-first` card on the right.

`usa-card__media--exdent`, `usa-card__header--exdent`, `usa-card__footer--exdent`: These `--exdent` variants extend the container out over the card border, which often looks neater for cards with light-colored borders.

Screen Shot 2020-04-27 at 3 22 06 PM

> **Above:** Standard media on the left, media exdent on the right

`usa-card__media--set-aspect`: This variant allows designers to set a fixed aspect ratio to the media object. The default aspect is `16x9` but this can be combined with the `add-aspect` utility to set a custom aspect.

**Note:** This does not work with flag variant cards since flag variant cards have a fixed image width. If you want to see a flag image at its original aspect, instead of filling the available space, add a `height-auto` utility to the `` element.

Screen Shot 2020-04-27 at 3 34 25 PM

> **Above:** Left, standard media. Middle, `usa-card__media--set-aspect`. Right, `usa-card__media--set-aspect.add-aspect-1x1`

Screen Shot 2020-04-27 at 4 57 04 PM

> **Above:** Flag cards. Left, standard image. Right, with a `height-auto` utility class on the `` element

`usa-card__media--inset`: This variant insets the image to the perimeter padding instead.

Screen Shot 2020-04-27 at 5 14 24 PM

> **Above:** Standard cards. Left, standard image. Middle, with `usa-card__media--inset`. Right, with `usa-card--header-top` and `usa-card__media--inset`

Screen Shot 2020-04-27 at 5 00 38 PM

> **Above:** Flag cards. Left, standard image. Right, with `usa-card__media--inset`

## Added settings
- `$theme-card-gap`: Set the gap between cards.
- `$theme-card-flag-image-width`: Set the width of the media object in flag cards.
- `$theme-card-padding-y`: Set the vertical padding between card elements.

## Changed settings
- `$theme-card-header-font-family` → `$theme-card-header-typeset`: Pass a list of family, size, and line-height tokens to typeset the heading.
- `$theme-card-margin` → `$theme-card-margin-bottom`: Changed for clarity. Note that this value will get `1` unit more that the specified value at mobile to facilitate scrolling.
- `$theme-card-padding` → `$theme-card-perimeter`: Now this only affects the perimeter padding, not the padding in-between elements (now `$theme-card-padding-y`)

## Removed settings
- `$theme-card-bg-color`: Use utilities instead.

- - -
**TODO**
- [ ] Add new settings to settings docs 
- [x] Test in IE 11 and across browsers

### CODE REVIEW

File: src/stylesheets/components/_card.scss [hunk 1]
```diff
@@ -1,27 +1,52 @@
...
-    @include u-height("full");
+/* TODO:
+ * Add a way to change the flag breakpoint as a modifier/class?
+ * Is card grid gap be different from other gaps?
+ * Fix flag and set-aspect
+ */
 
-    .usa-card__body {
-      @include u-flex("fill");
-      flex-basis: auto; // IE 11
```
Comment (mejiaj) [inline src/stylesheets/components/_card.scss:9]: Removing this causes issues in IE11.

image

image

File: src/components/card/card--test.njk [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        </div>
+        <div class="usa-card__body">
+          <p>There is a quality even meaner than outright ugliness or disorder, and this meaner quality is the dishonest mask of pretended order, achieved by ignoring or suppressing the real order that is struggling to exist and to be served.</p>
+        </div>
+        <div class="usa-card__footer">
+          <a href="javascript:void(0);" class="usa-button">Pursue a better life</a>
+        </div>
+      </div>
+    </li>
+    <li class="usa-card tablet:grid-col-6 desktop:grid-col-fill">
```
Comment (mejiaj) [inline src/components/card/card--test.njk:108]: Media is getting stretched in IE11.

image

File: src/stylesheets/components/_card.scss [hunk 3]
```diff
@@ -31,127 +56,275 @@
...
 
 .usa-card__heading {
-  @include typeset($theme-card-header-font-family, "lg", 2);
+  @include typeset($theme-card-header-typeset);
   @include u-margin(0);
 }
 
+// Media
+// ---------------------------------
+.usa-card__media {
```
Comment (mejiaj) [inline src/stylesheets/components/_card.scss:75]: Image is popping out in IE11.

image

File: src/stylesheets/components/_card.scss [hunk 4]
```diff
@@ -31,127 +56,278 @@
...
+  .usa-card__media--inset {
+    @include u-padding-top(0);
   }
-}
 
-.usa-card--flag .usa-card__flag-content {
-  @include at-media($theme-card-flag-min-width) {
-    @include grid-col(8);
+  .usa-card__media {
+    @include u-order("initial");
```
Comment (mejiaj) [inline src/stylesheets/components/_card.scss:203]: IE11 doesn't support `initial`. We should use `inherit` or any other value besides `unset/initial`.

**Comment by mejiaj** [comment]
@thisisdano pushed up some IE11 fixes, please take a look and let me know if these work for you.

**Comment by thisisdano** [comment]
@mejiaj This looks great in IE!

File: src/components/card/_card--kitchen-sink.njk [hunk 5]
```diff
@@ -0,0 +1,747 @@
+<style>
```
Comment (mejiaj) [inline src/components/card/_card--kitchen-sink.njk:1]: Do we still need this file?

File: src/components/card/card--test.njk [hunk 6]
```diff
@@ -0,0 +1,193 @@
+<section class="">
```
Comment (mejiaj) [inline src/components/card/card--test.njk:1]: Do we need the empty `class=""`?

File: src/components/test/kitchen-sink.config.yml [hunk 7]
```diff
@@ -34,6 +34,9 @@ context:
       - text: Yet another page
 
   inner:
+    - title: Cards
+      id: cards
+      component: card--kitchen-sink
```
Comment (mejiaj) [inline src/components/test/kitchen-sink.config.yml:39]: Do we still need card--kitchen-sink now that we've defined the markup?

File: src/stylesheets/settings/_settings-utilities.scss [hunk 8]
```diff
@@ -17,7 +17,7 @@ https://designsystem.digital.gov/utilities
 ----------------------------------------
 */
 
-$utilities-use-important: false !default;
+$utilities-use-important: true !default;
```
Comment (mejiaj) [inline src/stylesheets/settings/_settings-utilities.scss:20]: Just double-checking. Was this for testing or will utilities be `!important` from now on?

**Comment by mejiaj** [comment]
@thisisdano 

Shouldn't the last 3 variable settings should be:
```scss
$theme-card-margin-bottom: 4 !default;
$theme-card-padding-perimeter: 3 !default;
$theme-card-padding-y: 2 !default;
```

Instead of
image

File: src/stylesheets/settings/_settings-utilities.scss (hunk 8 above)
Comment (thisisdano) [reply src/stylesheets/settings/_settings-utilities.scss:20]: Just for testing, I'll revert

**Pull request merged**